MapleServer2/PacketHandlers/Game/FishingHandler.cs

```diff
@@ -1,47 +1,502 @@
+                    }
+                    checkBlock.X -= Block.BLOCK_SIZE;
+                    checkBlock.Y = startCoord.Y; // reset Y
+                }
+            }
+            else if (direction == Direction.SOUTH_WEST)
+            {
+                checkBlock.Y -= (2 * Block.BLOCK_SIZE); // start at the corner
+
+                for (int yAxis = 0; yAxis < 5; yAxis++)
+                {
+                    for (int xAxis = 0; xAxis < 3; xAxis++)
+                    {
+                        checkBlock.X -= Block.BLOCK_SIZE;
+                        for (int zAxis = 0; zAxis < 3; zAxis++)
+                        {
+
+                            if (MapMetadataStorage.BlockAboveExists(mapId, checkBlock.ToShort()))
+                            {
+                                break;
+                            }
+
+                            MapBlock block = MapMetadataStorage.GetMapBlock(mapId, checkBlock.ToShort());
+                            if (block == null || !IsLiquidBlock(block))
+                            {
+                                checkBlock.Z -= Block.BLOCK_SIZE;
+                                continue;
+                            }
+
+                            blocks.Add(block);
+                            checkBlock.Z -= Block.BLOCK_SIZE;
+                        }
+                        checkBlock.Z = startCoord.Z; // reset Z
+                    }
+                    checkBlock.Y += Block.BLOCK_SIZE;
+                    checkBlock.X = startCoord.X; // reset X
+                }
+            }
+            else if (direction == Direction.SOUTH_EAST)
+            {
+                checkBlock.X -= (2 * Block.BLOCK_SIZE); // start at the corner
+
+                for (int xAxis = 0; xAxis < 5; xAxis++)
+                {
+                    for (int yAxis = 0; yAxis < 3; yAxis++)
+                    {
+                        checkBlock.Y -= Block.BLOCK_SIZE;
+                        for (int zAxis = 0; zAxis < 3; zAxis++)
+                        {
+
+                            if (MapMetadataStorage.BlockAboveExists(mapId, checkBlock.ToShort()))
+                            {
+                                break;
+                            }
+
+                            MapBlock block = MapMetadataStorage.GetMapBlock(mapId, checkBlock.ToShort());
+                            if (block == null || !IsLiquidBlock(block))
+                            {
+                                checkBlock.Z -= Block.BLOCK_SIZE;
+                                continue;
+                            }
+
+                            blocks.Add(block);
+                            checkBlock.Z -= Block.BLOCK_SIZE;
+                        }
+                        checkBlock.Z = startCoord.Z; // reset Z
+                    }
+                    checkBlock.X += Block.BLOCK_SIZE;
+                    checkBlock.Y = startCoord.Y; // reset Y
```

Comment:
Could put this inside a function so it gets reused

Reply:
I'm not exactly sure how. If the player faces a different direction, X and Y values swap and so does the operator from - to + (vice versa).

Reply:
for for loop for Z is the same for all the four cases